### DISCUSSION TITLE
Applying affine transform to streamlines in a SFT object

### DISCUSSION THREAD

**Discussion by arokem** [dipy/dipy#2078] [Registration] [answered]
The SFT object has methods for moving streamlines between different spaces. But, what if we want to move the streamlines, based on some other affine transform? For example, if the same subject were scanned twice, and we wanted to move the streamlines from experiment 1 to be in register with experiment 2. It seems that we would have to call `transform_tracking_output` on the streamlines stored in SFT and then initialize another SFT object. Or am I doing this wrong?

**Comment by arokem** [comment]
@frheault : this seems like something you might know the answer to. Is there anything we're missing here?

**Comment by frheault** [comment, marked as answer]
Your intuition is right about recreating a SFT!

I would strongly advise registering the two images (or two tractogram, using SLR) first.
Once you have that transformation (warning: registration of img_1->img_2 give the inverse affine of tractogram_1->tractogram_2 using SLR) you can apply it to the streamlines in RASMM (using `transform_streamlines` is fine).
These modified streamlines should then be used to create a new StatefulTractogram where the reference would now be img_2 (target) or img_1_moved and **NOT** the original img_1 

It is crucial that the final tractogram has the same affine as img_1_moved (which incidentally is the one of img_2). Saving the transformed tractogram with its old header would be invalid.

 The results should be that tractogram_1_moved is aligned with img_1_moved and img_2 (and all have the same affine). And they should all overlap in MI-Brain or Dipy Horizon (RASMM) and if you are using TRK trying TrackVis is a good sanity check too

**Comment by arokem** [comment]
Great. Thanks for all these tips. Very useful. (cc: @36000)

Closing.
